src/main/java/org/hawkular/client/inventory/jaxrs/handlers/BulkCreateHandler.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.client.inventory.jaxrs.handlers;
+
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.hawkular.client.inventory.model.ElementType;
+
+/**
+ * Bulk Create API
+ * http://www.hawkular.org/docs/rest/rest-inventory.html#_bulk_create
+ */
+@Path("/hawkular/inventory/bulk")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public interface BulkCreateHandler {
+
+    @POST
+    @Path("/")
+    Response create(Map<String, Map<ElementType, List<Object>>> entities);
```

Comment:
@garethahealy bulk create supports for a query parameter `at`
I see most of the api has `at` query parameter on api document http://www.hawkular.org/docs/rest/rest-inventory.html